modules/thermo-cycler/production/eepromWriter/eepromWriter.ino

```diff
@@ -0,0 +1,188 @@
+#include <Wire.h>
+#include <ot_shared_data.h>
+#include <Arduino.h>
+
+#define WP_PIN                  26
+#define PIN_FAN_SINK_CTRL       A4   // uses PWM frequency generator
+#define PIN_FAN_SINK_ENABLE     2    // Heat sink fan
+#define PIN_FRONT_BUTTON_LED    24
+#define PIN_PELTIER_ENABLE      7
+#define PIN_MOTOR_CURRENT_VREF  A0
+
+String serial_num, model_num;
+bool reading_in_serial = false;
+
+bool eeprom_write_char(byte word_address, byte value)
+{
+  Wire.beginTransmission(OT_EEPROM_ADDR);
+  Wire.write(word_address);
+  Wire.write(value);
+  byte error = Wire.endTransmission();
+  delay(OT_WR_TIME);
+  if (error)
+  {
+    return false;
+  }
+  return true;
+}
+
+char eeprom_read_char(byte word_address)
+{
+  Wire.beginTransmission(OT_EEPROM_ADDR);
+  Wire.write(word_address);
+  byte error = Wire.endTransmission();
+  if (!error)
+  {
+    Wire.requestFrom(OT_EEPROM_ADDR, 1); // request 1 byte from eeprom
+    if (Wire.available())    // slave may send less than requested
+    {
+      char c = Wire.read(); // receive a byte as character
+      return c;         // print the character
+    }
+  }
+  return '~';
+}
+
+bool write_to_eeprom()
+{
+  uint8_t serial_len = serial_num.length();
+  uint8_t model_len = model_num.length();
+  char serial_arr[serial_len];
+  serial_num.toCharArray(serial_arr, serial_len+1);
```

Comment:
Why `serial_len+1`? won't that overrun the buffer? if you want room for the null byte make the buffer bigger too